samples/dnn/tf_text_graph_common.py

```diff
@@ -270,7 +270,10 @@ def removeIdentity(graph_def):
     identities = {}
     for node in graph_def.node:
         if node.op == 'Identity' or node.op == 'IdentityN':
-            identities[node.name] = node.input[0]
+            if node.input[0] in identities:
+                identities[node.name] = identities[node.input[0]]
+            else:
+                identities[node.name] = node.input[0]
```

Comment:
I think it make sense to make it more generic:
```python
inp = node.input[0]
while inp in identities:
  inp = identities[inp]
identities[node.name] = inp
```

Reply:
@dkurt, thank you for the review! 
However, is there a possibility of random graph traversal or graph traversal from child nodes to parent or some particular Identity sequence case? If `for node in graph_def.node:` determines descending traversal from a parent node, then we don't need an extra `while inp in identities:` loop because we sequentially handle nodes and if there are two or more nodes of `Identity` type (ex.: a -> identity_0 [inp: a] -> identity_1  [inp: identity_0]  -> ... -> identity_n [inp: identity_n-1] -> b [inp: identity_n]), we will check the presence of current identity node input in `identities = {}` and if it's there, we will add the current identity node to the  `identities = {}`, overwriting its input value with the input value of initial identity.

Reply:
At this moment nodes order is sorted by execution (by `writeTextGraph` method).